app/test/testmerginapi.cpp

```diff
@@ -1009,12 +1010,67 @@ void TestMerginApi::testConflictRemoteAddLocalAdd()
   uploadRemoteProject( mApi, mUsername, projectName );
 
   // verify the result: the server version should be in test1.txt
-  // and the local version should go to test1.txt_conflict_<username>_<version>
-  QString conflictFilename = filename + "_conflict_" + mUsername + "_v1";
+  // and the local version should go to conflicted copy file
+  QString conflictFilename = projectDir + "/test-new-file (conflicted copy, " + mUsername + " v1).txt";
   QCOMPARE( readFileContent( filename ), QByteArray( "new remote content" ) );
   QCOMPARE( readFileContent( conflictFilename ), QByteArray( "new local content" ) );
 }
 
+void TestMerginApi::testEditConflictScenario()
+{
+  // this test simulates creation of edit conflict when two
+  // clients are trying to update the same attribute.
+  // edit conflict file should be created inside project folder and synced to server
+
+  QString projectName = "testEditConflictScenario";
+  QString projectDir = mApi->projectsPath() + "/" + projectName;
+  QString extraProjectDir = mApiExtra->projectsPath() + "/" + projectName;
+
+  // folder rebase_edit_conflict
+  QString dataProjectDir = TestUtils::testDataDir() + "/" + QStringLiteral( "rebase_edit_conflict" );
+  qDebug() << "About to create the project";
+  createRemoteProject( mApiExtra, mUsername, projectName, mTestDataPath + "/" + TEST_PROJECT_NAME + "/" );
+  qDebug() << "Project has been created!";
+  QString dbName = QStringLiteral( "data.gpkg" );
+  QString baseDB = dataProjectDir + QStringLiteral( "/base.gpkg" );
+  QString localChangeDB = dataProjectDir + QStringLiteral( "/local-change.gpkg" );
+  QString remoteChangeDB = dataProjectDir + QStringLiteral( "/remote-change.gpkg" );
+
+  downloadRemoteProject( mApi, mUsername, projectName );
+
+  // upload base db
+  InputUtils::copyFile( baseDB, projectDir + "/" + dbName );
+  uploadRemoteProject( mApi, mUsername, projectName );
+
+  // both clients now sync the project so that both of them have base gpkg
+  downloadRemoteProject( mApi, mUsername, projectName );
+  downloadRemoteProject( mApiExtra, mUsername, projectName );
+
+  // both clients now make change to the same field
+  InputUtils::removeFile( projectDir + "/" + dbName );
+  InputUtils::removeFile( extraProjectDir + "/" + dbName );
+  InputUtils::copyFile( localChangeDB, projectDir + "/" + dbName );
+  InputUtils::copyFile( remoteChangeDB, extraProjectDir + "/" + dbName );
+
+  // client B syncs his changes
+  uploadRemoteProject( mApiExtra, mUsername, projectName );
+
+  //
+  // now client A syncs, resulting in edit conflict
+  //
+
+  uploadRemoteProject( mApi, mUsername, projectName );
+
+  QDir projDir( projectDir );
+
+  // check the edit conflict file presence
+  QVERIFY( InputUtils::fileExists( projectDir + "/" + QString( "data (edit conflict, %1 v2).gpkg" ).arg( mUsername ) ) );
```

Comment:
I think edit conflicts should always have `.json` suffix even though they are related to a `.gpkg` file - otherwise people may get confused and try to load edit conflict files as geopackages...